lib/stripe/core_resources/setup_intent.ex

```diff
@@ -0,0 +1,187 @@
+    |> make_request()
+  end
+
+  @doc """
+  Retrieves the details of a SetupIntent that has previously been created.
+  See the [Stripe docs](https://stripe.com/docs/api/setup_intents/retrieve).
+  """
+  @spec retrieve(Stripe.id() | t, params, Stripe.options()) ::
+          {:ok, t} | {:error, Stripe.Error.t()}
+        when params:
+               %{
+                 optional(:client_secret) => String.t()
+               }
+               | %{}
+  def retrieve(id, params, opts \\ []) do
+    new_request(opts)
+    |> put_endpoint(@plural_endpoint <> "/#{get_id!(id)}")
+    |> put_params(params)
+    |> put_method(:get)
+    |> make_request()
+  end
+
+  @doc """
+  Updates a SetupIntent object.
+  See the [Stripe docs](https://stripe.com/docs/api/setup_intents/update).
+  """
+  @spec update(Stripe.id() | t, params, Stripe.options()) :: {:ok, t} | {:error, Stripe.Error.t()}
+        when params:
+               %{
+                 optional(:customer) => Stripe.id() | Stripe.Customer.t(),
+                 optional(:description) => String.t(),
+                 optional(:metadata) => map,
+                 optional(:payment_method) => Stripe.id(),
+                 optional(:payment_method_types) => [String.t()]
+               }
+               | %{}
+  def update(id, params, opts \\ []) do
+    new_request(opts)
+    |> put_endpoint(@plural_endpoint <> "/#{get_id!(id)}")
+    |> put_method(:post)
+    |> put_params(params)
+    |> make_request()
+  end
+
+  @doc """
+  Confirm that your customer intends to set up the current or provided payment method.
+  See the [Stripe docs](https://stripe.com/docs/api/setup_intents/confirm).
+  """
+  @spec confirm(Stripe.id() | t, params, Stripe.options()) ::
+          {:ok, t} | {:error, Stripe.Error.t()}
+        when params:
+               %{
+                 optional(:payment_method) => Stripe.id(),
+                 optional(:return_url) => String.t()
+               }
+               | %{}
+  def confirm(id, params, opts \\ []) do
+    new_request(opts)
+    |> put_endpoint(@plural_endpoint <> "/#{get_id!(id)}" <> "/confirm")
+    |> put_method(:post)
+    |> put_params(params)
+    |> make_request()
+  end
+
+  @doc """
+  A SetupIntent object can be canceled when it is in one of these statuses:
+  `requires_payment_method`, `requires_capture`, `requires_confirmation`, `requires_action`.
+  See the [Stripe docs](https://stripe.com/docs/api/setup_intents/cancel).
+  """
+  @spec cancel(Stripe.id() | t, params, Stripe.options()) :: {:ok, t} | {:error, Stripe.Error.t()}
+        when params:
+               %{
+                 optional(:cancellation_reason) => String.t()
+               }
+               | %{}
+  def cancel(id, params, opts \\ []) do
+    new_request(opts)
+    |> put_endpoint(@plural_endpoint <> "/#{get_id!(id)}" <> "/cancel")
+    |> put_method(:post)
+    |> put_params(params)
+    |> make_request()
+  end
+end
```

Comment:
`def list` is an option as well!  https://stripe.com/docs/api/setup_intents/list